ballerina/tests/mock_service.bal

```diff
@@ -0,0 +1,352 @@
+        return {
+            "schedule_id": scheduleId,
+            "summary": "Test Schedule",
+            "description": "A test schedule",
+            "duration": 30,
+            "capacity": 1,
+            "active": true,
+            "status": "confirmed",
+            "availability_override": false
+        };
+    }
+
+    # Delete schedules
+    #
+    # + scheduleId - The unique identifier of the schedule.
+    # + user_id - User ID query parameter
+    # + return - returns can be any of following types
+    # http:NoContent (If successful, this method returns an empty response body.)
+    # http:Response (The request has failed.)
+    resource function delete schedules/[string scheduleId](string? user_id) returns http:Response {
+        http:Response response = new;
+        response.statusCode = 204;
+        return response;
+    }
+
+    # Patch schedules
+    #
+    # + scheduleId - The schedule's unique identifier.
+    # + user_id - User ID query parameter
+    # + return - returns can be any of following types
+    # http:NoContent (If successful, this method returns an empty response body.)
+    # http:Response (The request has failed.)
+    resource function patch schedules/[string scheduleId](@http:Payload json payload, string? user_id) returns http:Response {
+        http:Response response = new;
+        response.statusCode = 204;
+        return response;
+    }
+
+    # Single use link
+    #
+    # + return - returns can be any of following types
+    # http:Created (If successful, this method returns a scheduling link URL.)
+    # http:Response (The request has failed.)
+    resource function post schedules/single_use_link(@http:Payload json payload) returns json|http:Response {
+        json schedulePayload = payload;
+        string scheduleId = schedulePayload is map<json> && schedulePayload.hasKey("schedule_id") && schedulePayload["schedule_id"] is string
+            ? <string>schedulePayload["schedule_id"] : "default-id";
+        return <json>{
+            "scheduling_url": "https://scheduler.zoom.us/single-use/" + scheduleId
+        };
+    }
+
+    # Get user
+    #
+    # + userId - The user ID
+    # + return - returns can be any of following types
+    # http:Ok (If successful, this method returns user information.)
+    # http:Response (The request has failed.)
+    resource function get users/[string userId]() returns json|http:Response {
+        return {
+            "display_name": "Test User",
+            "scheduling_url": "https://scheduler.zoom.us/test-user",
+            "time_zone": "UTC",
+            "slug": "test-user"
+        };
+    }
+};
+
+function init() returns error? {
+    if isLiveServer {
+        log:printInfo("Skipping mock server initialization as the tests are running on live server");
+        return;
+    }
+    log:printInfo("Initiating mock server");
+    check httpListener.attach(mockService, "/");
+    check httpListener.'start();
+}
```

Comment:
Keep an empty line at the end of the file. Configure your IDE to automatically add this empty line when saving a file.

```suggestion
}

```